src/simulation/thermal/sensorThermal/sensorThermal.h

```diff
@@ -57,31 +67,44 @@ class SensorThermal: public SysModel {
     Message<TemperatureMsgPayload> temperatureOutMsg; //!< output temperature message
 
     Eigen::Vector3d nHat_B;                     //!< [-] Sensor normal unit vector relative to the spacecraft body frame.
+    TempFaultState_t faultState;                 //!< [-] Fault status variable
 
     double sensorPowerDraw;                     //!< [W] Power consumed by the sensor (+).
-    uint64_t sensorStatus;                        //!< [-] Sensor status (0/1)
+    uint64_t sensorStatus;                      //!< [-] Sensor status (0/1)
```

Comment:
:small_red_triangle_down: It looks like this toggle is used specifically to check whether a message payload was read/received. Should the user ever been setting or interacting with this variable?  Reading the code I expect not. If this is the case the variable should be made private to ensure it's not available to the unwitting user. Also it might benefit from a more descriptive name, because upon first read I inferred that it indicated something about the sensors on/off status. But I was wrong. Ideally, one wouldn't need to read the source code to have a good understanding of what value/purpose the variable holds.

Reply:
I think we need input from Adam about the role of sensorStatus. My only change here was to align the comment; he wrote this part of the module. The message can change the 0/1 state, but it is still used in line 267 as part of the environmental "heating up" calculation. I don't know why you would have nonzero sensorPowerDraw but then also want a zero sensorStatus though.

Reply:
If the user wants to spin up a sim where there is a thermal model of the sensor (this module) and a power model of the sensor that connects to a battery, then a deviceStatus message can be connected to both modules to turn the sensor on or off (when I say "turn off" I mean turn off power consumption, so Pat may be onto something with a more descriptive name). In that case you can have a nonzero sensorPowerDraw, but turn the device on/off as you wish with the deviceStatus message (which can be an output from another module or written directly in Python). Currently, readMessages automatically sets sensorStatus equal to one if no message is connected. Even if the user changes this to zero manually, it still gets set to one. Therefore, whether or not its public/private is currently irrelevant as it's overwritten. 

Upon reflection, I don't think we want this to get set to one every time readMessages is called. It should only be set to 1 in the constructor and should remain public. This allows the user to just set the sensorStatus directly at the Python level WITHOUT having to create a deviceStatus message. This is what I commonly do with the power and data buffer modules in the Gym environments, and I believe this module should function similarly. 

So, to respond directly to Pat, I do want the user to be able to set this, but I think you're correct in that it needs a more descriptive name. Documentation could probably be updated to make this more clear as well. Good catch!

Reply:
What updates should be made to the variable name and documentation?